### ISSUE TITLE
Pre-Commit for JS mapping files introduces wrong EOL characters

### ISSUE THREAD

**Issue by JoergAtGithub** [mixxxdj/mixxx#12974]
### Bug Description

If I commit a JavaScript mapping file with changes and pre-commit modifies it, it does not use the EOL characters as written by the Git client at checkout. It seems to use always Unix EOL for the modified lines. In the next run it complains about the mixed line end characters in the file.

I'm on Windows and my `.gitconfig` file contains the unmodified Git default setting of `autocrlf = true`

### Version

2.4

### OS

Windows 11

**Comment by Swiftb0y** [comment]
Weird, sounds why is this only happening for js files? Do you know which tool/hook is responsible?

**Comment by JoergAtGithub** [comment]
I think this is due to #12400. Now only the changed lines are modified by eslint. These get another EOL character than the files checked out by git and than the mixed EOL character check fails at th next git operation.

**Comment by Swiftb0y** [comment]
Do I understand correctly that your hypothesis is that eslint introduces EOL characters which are different than the rest of the file in only the changed lines?

**Comment by JoergAtGithub** [comment]
Yes, eslint changes the EOL characters, but in the modified lines only.

**Comment by Swiftb0y** [comment]
I see.

**Comment by Swiftb0y** [comment]
https://eslint.org/docs/latest/rules/linebreak-style#using-this-rule-with-version-control-systems

But I guess that's not really desired? Should we instead remove the rule outright?

**Comment by JoergAtGithub** [comment]
Maybe we can introduce a postprocessing after eslint

**Comment by Swiftb0y** [comment]
Pre-commit should be able to do that easily, but wouldn't that fight with the postprocessing done by git again?

**Comment by Swiftb0y** [comment]
![image](https://github.com/mixxxdj/mixxx/assets/12380386/0a1ade4a-1484-4fe7-ab68-efe69615e724)

I guess we can only set it to `auto` fix?
